go/store/nbs/journal_chunk_source.go

```diff
@@ -209,6 +209,65 @@ func (s journalChunkSource) close() error {
 	return nil
 }
 
+func (s journalChunkSource) iterateAllChunks(ctx context.Context, cb func(chunks.Chunk)) error {
+	var finalErr error
+
+	s.journal.lock.RLock()
+	defer s.journal.lock.RUnlock()
```

Comment:
Maybe a TODO that should probably be a call to `processJournalRecords` with a Reader and size which comes from `s.journal.snapshot()`, just filtering for the `chunkJournalRecKind` and giving the payload to NewCompressedChunk?